src/lib/formValidator.js

```diff
@@ -0,0 +1,103 @@
+import { monthFullNames } from '../modules/Experiences/AddOrEditExperienceForm/staticData';
+
+const experienceValidator = (values) => {
+  let isError = false;
+  let err = {};
+
+  if (values.host.length < 1) {
+    isError = true;
+    err.host_error = 'Required';
+  }
+  if (values.degree !== undefined) {
+    if (values.degree.length < 1) {
+      isError = true;
+      err.degree_error = 'Required';
+    }
+  }
+  if (values.title.length < 1) {
+    isError = true;
+    err.title_error = 'Required';
+  }
+  if (!values.start_month) {
+    isError = true;
+    err.startMonth_error = 'Required';
+  }
+  if (!values.start_year) {
+    isError = true;
+    err.startYear_error = 'Required';
+  }
+  if (!values.end_month) {
+    isError = true;
+    err.endMonth_error = 'Required';
+  }
+  if (!values.end_year) {
+    isError = true;
+    err.endYear_error = 'Required';
+  }
+
+  if (values.start_month && values.start_year && values.end_month && values.end_year) {
+    if (values.end_year === values.start_year) {
+      if (monthFullNames.indexOf(values.end_month) < monthFullNames.indexOf(values.start_month)) {
+        isError = true;
+        err.endMonth_error = 'End month must be later than start month';
+      }
+    }
+  }
+  if (values.start_year && values.end_year) {
+    if (values.end_year < values.start_year) {
+      isError = true;
+      err.endYear_error = 'End year must be greater than start year';
+    }
+  }
+
+  return { isError, err };
+};
+
+const newProfileValidator = (values) => {
+  let isError = false;
+  let err = {};
+
+  if (values.first_name === undefined || values.first_name.length < 1) {
+    isError = true;
+    err.firstName_error = 'Required';
+  }
+
+  if (values.last_name === undefined || values.last_name.length < 1) {
+    isError = true;
+    err.lastName_error = 'Required';
+  }
+
+  if (values.email === undefined) {
+    isError = true;
+    err.email_error = 'Required';
+  } else if (!validateEmail(values.email)) {
+    isError = true;
+    err.email_error = 'Invalid email address';
+  }
+  if (values.phone_primary === undefined || values.phone_primary.length < 1) {
+    isError = true;
+    err.phonePrimary_error = 'Required';
+  } else if (isNaN(parseInt(values.phone_primary))) {
+    isError = true;
+    err.phonePrimary_error = 'Numbers only';
+  } else if (values.phone_primary.length !== 10) {
+    isError = true;
+    err.phonePrimary_error = 'Phone number must be 10 digits';
+  }
+
+  return { isError, err };
+};
+
+// Validate RegEx
+const validateEmail = (inputText) => {
+  const mailFormat = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/;
```

Comment:
Emails are notoriously hard to validate, largely because there's no broad agreement on exactly what should or shouldn't be allowed in an email address (despite a number of RFCs that have tried, most recently [RFC5322](https://tools.ietf.org/html/rfc5322#section-3.4)). Probably in the long run we should use a high quality library, but in the short term I suggest we make use of other work people have done on this:
```regex
/\b[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}\b/i
```
This is from here: https://www.regular-expressions.info/email.html, which is also a great article on some of the complexities that can come up when trying to validate emails and why you might not want to use a RFC5322 based regex.

(Also see https://en.wikipedia.org/wiki/Email_address#Examples for some other examples of stuff that RFC5322 allows that we probably don't want to actually handle)

Reply:
Your RegEx gives a pass to this email "sdas@fs@f.com" (cannot validate double @@)

Reply:
Oops, I mistranslated from the website, good catch. I think this should be correct, though you should still test it (just changed from word boundaries to match the whole string)
```
/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i
```